src/routes/v3/oauth_clients.rs

```diff
@@ -0,0 +1,387 @@
+    if let Some(client) = client {
+        current_user.validate_can_interact_with_oauth_client(client.created_by.into())?;
+        OAuthClient::remove(client.id, &**pool).await?;
+
+        Ok(HttpResponse::NoContent().body(""))
+    } else {
+        Ok(HttpResponse::NotFound().body(""))
+    }
+}
+
+#[derive(Serialize, Deserialize, Validate)]
+pub struct OAuthClientEdit {
+    #[validate(
+        custom(function = "crate::util::validate::validate_name"),
+        length(min = 3, max = 255)
+    )]
+    pub name: Option<String>,
+
+    #[validate(
+        custom(function = "crate::util::validate::validate_url"),
+        length(max = 255)
+    )]
+    pub icon_url: Option<Option<String>>,
+
+    pub max_scopes: Option<Scopes>,
+
+    #[validate(length(min = 1))]
+    pub redirect_uris: Option<Vec<String>>,
+}
+
+#[patch("oauth_app/{id}")]
+pub async fn oauth_client_edit(
+    req: HttpRequest,
+    client_id: web::Path<String>,
+    client_updates: web::Json<OAuthClientEdit>,
+    pool: web::Data<PgPool>,
+    redis: web::Data<RedisPool>,
+    session_queue: web::Data<AuthQueue>,
+) -> Result<HttpResponse, ApiError> {
+    let current_user = get_user_from_headers(
+        &req,
+        &**pool,
+        &redis,
+        &session_queue,
+        Some(&[Scopes::OAUTH_CLIENT_WRITE]),
+    )
+    .await?
+    .1;
+
+    client_updates
+        .validate()
+        .map_err(|e| ApiError::Validation(validation_errors_to_string(e, None)))?;
+
+    if client_updates.icon_url.is_none()
+        && client_updates.name.is_none()
+        && client_updates.max_scopes.is_none()
+    {
+        return Err(ApiError::InvalidInput("No changes provided".to_string()));
+    }
+
+    if let Some(existing_client) =
+        get_oauth_client_from_str_id(client_id.into_inner(), &**pool).await?
+    {
+        current_user.validate_can_interact_with_oauth_client(existing_client.created_by.into())?;
+
+        let mut updated_client = existing_client.clone();
+        let OAuthClientEdit {
+            name,
+            icon_url,
+            max_scopes,
+            redirect_uris,
+        } = client_updates.into_inner();
+        if let Some(name) = name {
+            updated_client.name = name;
+        }
+
+        if let Some(icon_url) = icon_url {
+            updated_client.icon_url = icon_url;
+        }
+
+        if let Some(max_scopes) = max_scopes {
+            updated_client.max_scopes = max_scopes;
+        }
+
+        let mut transaction = pool.begin().await?;
+        updated_client
+            .update_editable_fields(&mut *transaction)
+            .await?;
+
+        if let Some(redirects) = redirect_uris {
+            edit_redirects(redirects, &existing_client, &mut transaction).await?;
+        }
+
+        transaction.commit().await?;
+
+        Ok(HttpResponse::Ok().body(""))
+    } else {
+        Ok(HttpResponse::NotFound().body(""))
+    }
+}
+
+#[get("user/oauth_authorizations")]
```

Comment:
also think the user prefix should be removed from this

Reply:
Will do